contracts/governance/Governor.sol

```diff
@@ -323,59 +371,55 @@ abstract contract Governor is Context, ERC165, EIP712, IGovernor, IERC721Receive
     }
 
     /**
-     * @dev See {IGovernor-execute}.
+     * @dev Internal execution mechanism. Can be overridden (with a super call) to add logic before or after the
+     * execution.
+     *
+     * Emits a {IGovernor-ProposalExecuted} event.
      */
-    function execute(
+    function _execute(
+        uint256 proposalId,
```

Comment:
In your explanation you say:

> you can call the internal version directly if you want to bypass the public checks

But one of the issues with this is that you can pass a `proposalId` that doesn't correspond to the proposal parameters. Is this a bug or a feature? What do you think?

Reply:
> But one of the issues with this is that you can pass a `proposalId` that doesn't correspond to the proposal parameters. Is this a bug or a feature? What do you think?

Tat is something I don't like, and I'd like to avoid, but I'm not sure how to do without re-hashing the proposalId. Are we ok with computing it twice ?

Reply:
Its true that currently `execute/_execute` and `queue/_queue` don't bring much in terms of features. I did it for clarity (same pattern everywhere), and to have clear guidelines as to what should be overriden depending on the usecase.

You want to add accesscontrol to queue and execute ? do it on the public one. You want to add internal tracking logic ? do it on the internal one.

I'm open to merging them if we fell that would be preferable.

Reply:
I think the "weakness" about `proposalId` is acceptable. But the separation doesn't seem to work well to add access control. In the case of cancel if we want to add a new way to cancel that uses different "access control" we would be violating the guidelines I'm proposing in https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4415.

In the case of cancel we could solve this by having a `_canCancel(msgSender, proposalId)` function. What do you think?

Reply:
When I said "add access control" I meant adding restriction. Propose / queue / execute are already very open (appart from the proposalThreshold part), and adding restriction is "easy".

The `_canCancel` is interresting ... but ideally it would take all the proposal details (maybe the rules depend on the contract being called).